linera-execution/src/execution_state_actor.rs

```diff
@@ -97,6 +101,63 @@ where
                 callback.respond(balance);
             }
 
+            OwnerBalance { owner, callback } => {
+                let balance = self.system.balances.get(&owner).await?.unwrap_or_default();
+                callback.respond(balance);
+            }
+
+            Transfer {
+                source,
+                destination,
+                amount,
+                signer,
+                callback,
+            } => {
+                let (execution_outcome, _) = self
+                    .system
+                    .execute_operation(
+                        // Temporary hack creating fake OperationContext. We'll need just the
+                        // authenticated signer in the operation
+                        Self::get_fake_operation_context(signer),
+                        SystemOperation::Transfer {
+                            owner: source,
+                            recipient: crate::system::Recipient::Account(destination),
+                            amount,
+                            user_data: UserData::default(),
+                        },
```

Comment:
Rather than creating a fake context and a `SystemOperation` that isn't actually in the block, maybe let's extract the `Transfer` part of `SystemExecutionStateView::execute_operation` into a separate method (`SystemExecutionStateView::transfer`?) instead.

Reply:
Ah, nice, sounds cleaner! Let me try